AsyncDisplayKit/ASCellNode.mm

```diff
@@ -117,15 +118,18 @@ - (void)layoutDidFinish
   _viewControllerNode.frame = self.bounds;
 }
 
-- (void)__setNeedsLayout
+- (void)displayNodeDidInvalidateSizeOldSize:(CGSize)oldSize
 {
-  CGSize oldSize = self.calculatedSize;
-  [super __setNeedsLayout];
-  
-  //Adding this lock because lock used to be held when this method was called. Not sure if it's necessary for
-  //didRelayoutFromOldSize:toNewSize:
-  ASDN::MutexLocker l(__instanceLock__);
-  [self didRelayoutFromOldSize:oldSize toNewSize:self.calculatedSize];
+  ASSizeRange constrainedSize = ASSizeRangeMake(CGSizeZero, CGSizeMake(CGRectGetWidth(self.bounds), CGFLOAT_MAX));
+  if (_interactionDelegate != nil) {
+    constrainedSize = [_interactionDelegate constrainedSizeForNode:self];
+  }
+  CGSize newSize = [self layoutThatFits:constrainedSize].size;
+    
+  if (CGSizeEqualToSize(oldSize, newSize) == NO) {
+    self.frame = {self.frame.origin, newSize};
```

Comment:
Unfortunately we cannot remove it at this time as neither `_ASTableViewCell` or `_ASCollectionViewCell` has as `layoutSubviews` implementation that updates the node to the current bounds